### DISCUSSION TITLE
New Python user need help

### DISCUSSION THREAD

**Discussion by Predator4578** [mathoudebine/turing-smart-screen-python#468] [Q&A] [answered]
Hi i have never used python before

Now i just want to try out this project if it works with my screen
I installed Python 312 "latest" windows release
i have now no idea how to contiue
https://github.com/mathoudebine/turing-smart-screen-python/wiki/System-monitor-:-how-to-start
here it says
3. Install Python dependencies

NOTE: it is recommended to create and use a Python virtual environment for this project because it needs specific versions of its dependencies. Install required Python modules for this project:
python3 -m pip install -r requirements.txt / python.exe -m pip install -r requirements.txt

i have no idea how to use python i entered those commands and i get errors
>>> python.exe -m pip install -r requirements.txt
  File "<stdin>", line 1
    python.exe -m pip install -r requirements.txt
                  ^^^
SyntaxError: invalid syntax

pls help me

**Comment by unaigonzalezz** [comment]
Open CMD (You can press WIN+R, and enter CMD, or search it), open the folder where you decompressed the files with "CD 'folder'" and on this folder, execute "python3 -m pip install -r requirements.txt".

**Comment by GaryStar** [comment]
try ```py.exe``` instead of python.exe

**Comment by Predator4578** [comment]
Thank u both a lot i was thinking i need to enter this into python CLI, now i got it using CMD was the point...

Now i have another issue when entering the command it seem to stuck at a specific dependency 

Ignoring numpy: markers 'python_version < "3.9"' don't match your environment
Ignoring GPUtil: markers 'python_version < "3.12"' don't match your environment
Collecting GPUtil@ git+https://github.com/mathoudebine/gputil.git@1.4.0-py3.12 (from -r requirements.txt (line 18))
  Cloning https://github.com/mathoudebine/gputil.git (to revision 1.4.0-py3.12) to c:\users\user\appdata\local\temp\pip-install-2kcwglrr\gputil_2f119227ce214bb2b87a0b9df4fe6877
  Running command git clone --filter=blob:none --quiet https://github.com/mathoudebine/gputil.git 'C:\Users\USER\AppData\Local\Temp\pip-install-2kcwglrr\gputil_2f119227ce214bb2b87a0b9df4fe6877'
  fatal: invalid branch name: init.defaultBranch =
  error: subprocess-exited-with-error

  × git clone --filter=blob:none --quiet https://github.com/mathoudebine/gputil.git 'C:\Users\USER\AppData\Local\Temp\pip-install-2kcwglrr\gputil_2f119227ce214bb2b87a0b9df4fe6877' did not run successfully.
  │ exit code: 128
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error


any idea how to Fix this one?

**Reply by shampoolegs** [reply]
[](https://www.youtube.com/watch?v=NzN-tQ5lkYc)
https://www.youtube.com/watch?v=NzN-tQ5lkYc
I knew nothing about python too and this tutorial helped get me up and running

**Reply by Predator4578** [reply, marked as answer]
Hi thanks your Answer

i have followed the guide and i get the same error sadly, 

when i exclude the GPutil from Requirements.txt it installs everything else Fine

# For Nvidia GPU on all platforms
# GPUtil~=1.4.0; python_version < "3.12"
# GPUtil is broken for Python 3.12+ and not maintained anymore: fetch it from a fork where it is fixed
# GPUtil @ git+https://github.com/mathoudebine/gputil.git@1.4.0-py3.12 ; python_version >= "3.12"

but not excluded (Comment out) it errors to instal this dependency

# For Nvidia GPU on all platforms
GPUtil~=1.4.0; python_version < "3.12"
# GPUtil is broken for Python 3.12+ and not maintained anymore: fetch it from a fork where it is fixed
GPUtil @ git+https://github.com/mathoudebine/gputil.git@1.4.0-py3.12 ; python_version >= "3.12"

=
Ignoring numpy: markers 'python_version < "3.9"' don't match your environment
Ignoring GPUtil: markers 'python_version < "3.12"' don't match your environment
Collecting GPUtil@ git+https://github.com/mathoudebine/gputil.git@1.4.0-py3.12 (from -r requirements.txt (line 18))
Cloning https://github.com/mathoudebine/gputil.git (to revision 1.4.0-py3.12) to c:\users\user\appdata\local\temp\pip-install-2kcwglrr\gputil_2f119227ce214bb2b87a0b9df4fe6877
Running command git clone --filter=blob:none --quiet https://github.com/mathoudebine/gputil.git 'C:\Users\USER\AppData\Local\Temp\pip-install-2kcwglrr\gputil_2f119227ce214bb2b87a0b9df4fe6877'
fatal: invalid branch name: init.defaultBranch =
error: subprocess-exited-with-error

× git clone --filter=blob:none --quiet https://github.com/mathoudebine/gputil.git 'C:\Users\USER\AppData\Local\Temp\pip-install-2kcwglrr\gputil_2f119227ce214bb2b87a0b9df4fe6877' did not run successfully.
│ exit code: 128
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

the GPutil i guess is needed for the GPU values, it seem to have a issue now the Fork or whatever

I went now manually to:
https://github.com/mathoudebine/gputil
and there it says i should type "pip install gputil"
wich has worked i think

**Reply by mathoudebine** [reply]
Hello, I think this is because you do not have Git program installed on your computer. Sorry I missed that in the install guide. You need to download and install Git from here:
https://git-scm.com/download/win
If you just use the command `pip install gputil` it will install an older version of gputil which is not compatible with latest Python 3.12
Pleas run `pip uninstall gputil` first to remove the older version, then try to install requirements again from requirements.txt

**Reply by Predator4578** [reply]
hi thanks for reply

i have git installed on my machine
i use Python 3.12.2 (current installer)
